packages/logs/lib/otlp/otlpSpanProcessor.ts

```diff
@@ -0,0 +1,116 @@
+import type { ReadableSpan, Span, SpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { BatchSpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import type { Attributes, Context } from '@opentelemetry/api';
+import { stringToHash, stringifyError } from '@nangohq/utils';
+import { logger } from '../utils.js';
+import { otlpRoutingAttributeKey } from './otlp.js';
+import type { RouteConfig } from './otlp.js';
+
+interface BatchSpanProcessorWithRouteHash {
+    processor: BatchSpanProcessor;
+    routeHash: number;
+}
+
+export class RoutingSpanProcessor implements SpanProcessor {
+    private processors = new Map<string, BatchSpanProcessorWithRouteHash>();
+
+    constructor(routes: RouteConfig[]) {
+        try {
+            for (const route of routes) {
+                const processor = new BatchSpanProcessor(
+                    new OTLPTraceExporter({
+                        url: `${route.routingEndpoint.replace(/\/$/, '')}/traces`, // trim trailing slash if present
+                        headers: route.routingHeaders
+                    })
+                );
+                const routeHash = stringToHash(JSON.stringify(route));
```

Comment:
you could store an `id:uuid` inside the object and forget about this hashing

Reply:
I prefer this solution because even with a `version:uuid` in the otlp_settings a comparison will need to happen anyway to decide to update the settings value or not since we don't want the version to change if the entered values are the same. It then save an extra attributes in the settings